packages/rum-core/src/domain/rumEventsCollection/view/viewMetrics/trackFirstHidden.ts

```diff
@@ -2,40 +2,36 @@ import type { RelativeTime } from '@datadog/browser-core'
 import { addEventListeners, DOM_EVENT } from '@datadog/browser-core'
 import type { RumConfiguration } from '../../../configuration'
 
-let trackFirstHiddenSingleton: { timeStamp: RelativeTime } | undefined
-let stopListeners: (() => void) | undefined
+export type FirstHidden = ReturnType<typeof trackFirstHidden>
 
 export function trackFirstHidden(configuration: RumConfiguration, eventTarget: Window = window) {
-  if (!trackFirstHiddenSingleton) {
-    if (document.visibilityState === 'hidden') {
-      trackFirstHiddenSingleton = {
-        timeStamp: 0 as RelativeTime,
-      }
-    } else {
-      trackFirstHiddenSingleton = {
-        timeStamp: Infinity as RelativeTime,
-      }
-      ;({ stop: stopListeners } = addEventListeners(
-        configuration,
-        eventTarget,
-        [DOM_EVENT.PAGE_HIDE, DOM_EVENT.VISIBILITY_CHANGE],
-        (event) => {
-          if (event.type === 'pagehide' || document.visibilityState === 'hidden') {
-            trackFirstHiddenSingleton!.timeStamp = event.timeStamp as RelativeTime
-            stopListeners!()
-          }
-        },
-        { capture: true }
-      ))
-    }
-  }
+  let timeStamp: RelativeTime
+  let stopListeners: () => void
```

Comment:
<!-- Suggestions are specific requests to improve the subject under review. It is assumed that we all want to do what's best, so these comments are never dismissed as “mere suggestions”, but are taken seriously.  -->
**💬 suggestion:** ‏type should be `(() => void) | undefined`